src/main/java/org/opentripplanner/routing/impl/GraphPathFinder.java

```diff
@@ -88,8 +90,11 @@ public List<GraphPath<State, Edge, Vertex>> getPaths(
       .of()
       .setHeuristic(new EuclideanRemainingWeightHeuristic(maxCarSpeed))
       .setSkipEdgeStrategy(
-        new DurationSkipEdgeStrategy(
-          preferences.maxDirectDuration().valueOf(request.journey().direct().mode())
+        new ComposingSkipEdgeStrategy<>(
+          new DurationSkipEdgeStrategy(
+            preferences.maxDirectDuration().valueOf(request.journey().direct().mode())
+          ),
+          new BatteryDistanceSkipEdgeStrategy(BatteryValidator::wouldBatteryRunOut)
```

Comment:
We only need this heuristic for rental searches. Can you only add it if there mode contains rental?

Reply:
We added a method called getSkipEdgeStrategy where we check if the mode includes renting and call the BatterySkipEdgeStrategy if true